core/Command/Integrity/CheckApp.php

```diff
@@ -44,13 +50,21 @@ protected function configure() {
 	protected function execute(InputInterface $input, OutputInterface $output): int {
 		$appid = $input->getArgument('appid');
 		$path = (string)$input->getOption('path');
-		$result = $this->checker->verifyAppSignature($appid, $path, true);
-		$this->writeArrayInOutputFormat($input, $output, $result);
-		if (count($result) > 0) {
-			$output->writeln('<error>' . count($result) . ' errors found</error>', OutputInterface::VERBOSITY_VERBOSE);
-			return 1;
+		if ($path === '') {
+			$path = $this->appLocator->getAppPath($appid);
+		}
+		if ($this->appManager->isShipped($appid) || $this->fileAccessHelper->file_exists($path . '/appinfo/signature.json')) {
+			// Only verify if the application explicitly ships a signature.json file
+			$result = $this->checker->verifyAppSignature($appid, $path, true);
+			$this->writeArrayInOutputFormat($input, $output, $result);
+			if (count($result) > 0) {
+				$output->writeln('<error>' . count($result) . ' errors found</error>', OutputInterface::VERBOSITY_VERBOSE);
+				return 1;
+			}
+			$output->writeln('<info>No errors found</info>', OutputInterface::VERBOSITY_VERBOSE);
+		} else {
+			$output->writeln('<info>App signature not found, skipping app integrity check</info>', OutputInterface::VERBOSITY_VERBOSE);
```

Comment:
I think yellow is better, and from my pov it should be printed by default:
```suggestion
			$output->writeln('<comment>App signature not found, skipping app integrity check</comment>');
```

Otherwise good in general